bazel/external_crates.bzl

```diff
@@ -8,14 +8,39 @@ to regenerate Cargo Bazel lockfiles.
 load("@rules_rust//crate_universe:defs.bzl", "crate", "crates_repository", "splicing_config")
 load("//bazel:fuzz_testing.bzl", "DEFAULT_RUSTC_FLAGS_FOR_FUZZING")
 
-def sanitize_external_crates(sanitizers_enabled):
-    FUZZING_ANNOTATION = [crate.annotation(rustc_flags = DEFAULT_RUSTC_FLAGS_FOR_FUZZING)] if sanitizers_enabled else []
+def sanitize_external_crates():
+    FUZZING_ANNOTATION_LIB = [
+        crate.annotation(
+            rustc_flags = crate.select(
+                [],
+                {
+                    # Only enable fuzzing on x86_64 because it fails for wasm32:
+                    "x86_64-unknown-linux-gnu": DEFAULT_RUSTC_FLAGS_FOR_FUZZING,
+                },
+            ),
+        ),
+    ]
+    FUZZING_ANNOTATION_BIN = [
+        crate.annotation(
+            gen_binaries = True,
+            rustc_flags = crate.select(
+                [],
+                {
+                    # Only enable fuzzing on x86_64 because it fails for wasm32.
+                    # Additionally ic-wasm generates binaries so we need to link libclang_rt.fuzzer-x86_64.a:
+                    "x86_64-unknown-linux-gnu": DEFAULT_RUSTC_FLAGS_FOR_FUZZING +
+                                                ["-Clink-arg=/usr/lib/llvm-18/lib/clang/18/lib/linux/libclang_rt.fuzzer-x86_64.a"],
```

Comment:
Can you briefly explain what this is for? Or did I miss it in the description?

Reply:
Yeah it's explained a bit in the comment above. The `ic-wasm` crate is not just a library but also generates binaries. These need to link with `libclang_rt.fuzzer-x86_64.a` otherwise you get undefined symbol errors at link time.